src/OrleansCodeGenerator/Utilities/SyntaxFactoryExtensions.cs

```diff
@@ -311,12 +313,21 @@ public static TypeParameterConstraintClauseSyntax[] GetTypeConstraintSyntax(this
                         parameterConstraints.Add(SyntaxFactory.ClassOrStructConstraint(SyntaxKind.StructConstraint));
                     }
 
+                    // Follow with the base class or interface constraints.
                     foreach (var genericType in genericParameter.GetGenericParameterConstraints())
                     {
+                        // If the "struct" constraint was specified, skip the corresponding "ValueType" constraint.
+                        if (genericType == typeof(ValueType))
```

Comment:
Apparently the `struct` constraint is expressed first by the flag above and then by the `ValueType` class here, the latter of which we must omit.